src/ansys/fluent/core/transcript.py

```diff
@@ -0,0 +1,68 @@
+import threading
+from typing import Callable, Optional
+
+from ansys.fluent.core.services.transcript import TranscriptService
+
+lock = threading.Lock()
+
+
+class Transcript:
+    """Encapsulates a Fluent Transcript streaming service."""
+
+    def __init__(self, channel, metadata):
+        self._channel = channel
+        self._metadata = metadata
+        self.transcript_service = TranscriptService(self._channel, self._metadata)
+        self._transcript_thread: Optional[threading.Thread] = None
+        self._transcript_callbacks = {}
+        self._transcript_callback_id = 0
+
+    def add_transcript_callback(self, callback_fn: Callable, keep_new_lines=False):
+        """Initiates a fluent transcript streaming depending on the
+        callback_fn.
+
+        For eg.: add_transcript_callback(print) prints the transcript on
+        the interpreter screen.
+        """
+        with lock:
+            self._transcript_callbacks[self._transcript_callback_id] = (
+                callback_fn,
+                keep_new_lines,
+            )
+            returned_callback_id = self._transcript_callback_id
+            self._transcript_callback_id = self._transcript_callback_id + 1
```

Comment:
You can also use `itertools.count()` and next .

Reply:
@ajain-work, Thank you for the suggestion. This has been implemented.